aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -355,6 +355,34 @@ static void polygamma_kernel(TensorIterator& iter, int64_t n) {
   }
 }
 
+static void heaviside_tensor_kernel(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "heaviside_tensor_cpu", [&]() {
+    cpu_kernel(iter, [=](scalar_t self, scalar_t val) -> scalar_t {
+       if (self > 0) {
+         return 1.0;
+       } else if (self < 0) {
+         return 0.0;
+       } else {
+         return val;
+       }
+    });
+  });
+}
+
+static void heaviside_scalar_kernel(TensorIterator& iter, Scalar val) {
```

Comment:
I don't think you need a separate kernel for a scalar val because TensorIterator will handle the broadcasting for you.

Reply:
Yes, we no longer need  a separate kernel for scalar val, deleted.